From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct rtable
Patch-mainline: never, kabi
References: kabi

In networking-stable-18_11_21, upstream commit
d52e5a7e7ca49457dd31fc8b42fb7c0d58a31221 (ipv4: lock mtu in fnhe when
received PMTU < net.ipv4.route.min_pmtu) changed rt_pmtu u32 in struct
rtable to two bitmasks of the same size. It made the kABI checker to
complain.

Given the size of the structure did not change and rt_pmtu is used only
internally, hide the change from the kABI checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/net/route.h |    4 ++++
 1 file changed, 4 insertions(+)

--- a/include/net/route.h
+++ b/include/net/route.h
@@ -63,8 +63,12 @@ struct rtable {
 	__be32			rt_gateway;
 
 	/* Miscellaneous cached information */
+#ifdef __GENKSYMS__
+	u32			rt_pmtu;
+#else
 	u32			rt_mtu_locked:1,
 				rt_pmtu:31;
+#endif
 
 	u32			rt_table_id;
 
